aiographql/client/client.py

```diff
@@ -92,118 +91,182 @@ def __init__(
-        self, request: GraphQLRequest, headers: Optional[Dict[str, str]] = None
-    ) -> GraphQLTransaction:
-        return await self.request(request, method=QueryMethod.post, headers=headers)
+        self,
+        request: GraphQLRequest,
+        headers: Optional[Dict[str, str]] = None,
+        operation: Optional[str] = None,
+        variables: Optional[Dict[str, Any]] = None,
+    ) -> GraphQLResponse:
+        """
+        Helper method that wraps `GraphQLClient.request` with method set as
+        `QueryMethod.POST`.
+        """
+        return await self.request(
+            request,
+            method=QueryMethod.post,
+            headers=headers,
+            operation=operation,
+            variables=variables,
+        )
 
     async def get(
-        self, request: GraphQLRequest, headers: Optional[Dict[str, str]] = None
-    ) -> GraphQLTransaction:
-        return await self.request(request, method=QueryMethod.get, headers=headers)
+        self,
+        request: GraphQLRequest,
+        headers: Optional[Dict[str, str]] = None,
+        operation: Optional[str] = None,
+        variables: Optional[Dict[str, Any]] = None,
+    ) -> GraphQLResponse:
+        return await self.request(
+            request,
+            method=QueryMethod.get,
+            headers=headers,
+            operation=operation,
+            variables=variables,
+        )
 
     async def query(
-        self, request: GraphQLRequest, headers: Optional[Dict[str, str]] = None
-    ) -> GraphQLTransaction:
-        return await self.request(request=request, headers=headers)
-
-    async def _subscribe(self, subscription: GraphQLSubscription):
-        async with aiohttp.ClientSession() as session:
-            async with session.ws_connect(self.endpoint) as ws:
-                await ws.send_json(data=subscription.connection_init_request)
-
-                subscription.callbacks.register(
-                    GraphQLSubscriptionEventType.CONNECTION_ACK,
-                    SimpleTriggerCallback(
-                        function=ws.send_json,
-                        data=subscription.connection_start_request,
-                    ),
-                )
-
-                try:
-                    async for msg in ws:  # type:  aiohttp.WSMessage
-                        if msg.type != aiohttp.WSMsgType.TEXT:
-                            if msg.type == aiohttp.WSMsgType.ERROR:
-                                break
-                            continue
-
-                        # noinspection PyTypeChecker,PyNoneFunctionAssignment
-                        event = subscription.create_event(msg.json())
-                        await subscription.handle(event=event)
-
-                        if subscription.is_stop_event(event):
-                            break
-                except (asyncio.CancelledError, KeyboardInterrupt):
-                    await ws.send_json(data=subscription.connection_stop_request)
+        self,
```

Comment:
Is there a reason to keep this method just to call `request` method?

Reply:
Good question. TBH I am not sure, I have been thinking of renaming `request` to be `query` instead.